upstairs/src/lib.rs

```diff
@@ -5951,34 +5951,8 @@ impl Upstairs {
             return Err(());
         }
 
-        /*
-         * Get the next ID for the guest work struct we will make at the
-         * end. This ID is also put into the IO struct we create that
-         * handles the operation(s) on the storage side.
-         */
-        let mut gw = self.guest.guest_work.lock().await;
-        let mut downstairs = self.downstairs.lock().await;
         let ddef = self.ddef.lock().await.get_def().unwrap();
 
-        // While we've got the lock, update our current backpressure
-        self.set_backpressure_with_downstairs(&downstairs);
-
-        /*
-         * Verify IO is in range for our region.  If not give up now and
-         * report error.
-         */
-        match ddef.validate_io(offset, data.len()) {
```

Comment:
This needs to be before `extent_from_offset`, otherwise that function could panic.

Reply:
Good catch, done in 0bfdceec15768579fc0f29b831c6e2ef945f5cf5